include/highfive/bits/H5Converter_misc.hpp

```diff
@@ -267,32 +267,126 @@ struct data_converter<boost::numeric::ublas::matrix<T>, void> {
 #endif
 
 #ifdef H5_USE_EIGEN
+//compute size for single Eigen Matrix
+template <typename T, int M, int N>
+inline size_t compute_total_size(const Eigen::Matrix<T,M,N>& matrix)
+{
+    return matrix.rows() * matrix.cols();
+}
+
+//compute size for  std::vector of Eigens
+template <typename T, int M, int N>
+inline size_t compute_total_size(const std::vector<Eigen::Matrix<T,M,N>>& vec)
+{
+    return std::accumulate(vec.begin(), vec.end(), size_t{0u},
+                           [](size_t so_far, const auto& v){return so_far + (size_t)v.rows() * (size_t)v.cols(); });
+}
+
+//compute total row size for std::vector of Eigens
+template <typename T, int M, int N>
+inline size_t compute_total_row_size(const std::vector<Eigen::Matrix<T,M,N>>& vec)
+{
+    return std::accumulate(vec.begin(), vec.end(), size_t{0u},
+                           [](size_t so_far, const auto& v){return so_far + (size_t)v.rows(); });
+}
+
+
 // apply conversion to eigen matrix
 template <typename T, int M, int N>
 struct data_converter<Eigen::Matrix<T, M, N>, void> {
 
     typedef typename Eigen::Matrix<T, M, N> MatrixTMN;
 
     inline data_converter(MatrixTMN&, DataSpace& space)
-        : _dims(space.getDimensions()) {
+        : _dims(space.getDimensions()) {    
         assert(_dims.size() == 2);
     }
 
     inline typename type_of_array<T>::type* transform_read(MatrixTMN& array) {
-        if (_dims[0] != M || _dims[1] != N) {
-            array.resize(_dims[0], _dims[1]);
+
+        if (_dims[0] != (size_t)array.rows() || _dims[1] != (size_t)array.cols()) {
+            array.resize((Eigen::Index)_dims[0], (Eigen::Index)_dims[1]);
         }
-        return array.data;
+        return array.data();
     }
 
     inline typename type_of_array<T>::type* transform_write(MatrixTMN& array) {
-        return array.data;
+        return array.data();
     }
 
     inline void process_result(MatrixTMN&) {}
 
     std::vector<size_t> _dims;
 };
+
+template <typename T, int M, int N>
+inline void vectors_to_single_buffer(const std::vector<Eigen::Matrix<T,M,N>>& vec,
+                                     const std::vector<size_t>& dims,
+                                     const size_t current_dim,
+                                     std::vector<T>& buffer) {
+
+    check_dimensions_vector(compute_total_row_size(vec), dims[current_dim], current_dim);
+    for (const auto& k : vec) {
+        std::copy(k.data(), k.data()+k.size(), std::back_inserter(buffer));
```

Comment:
Regarding what is done above, it seems the number of inserted elements is known in advance. Maybe `std::vector::reserve` could be called accordingly before pushbacking.

Reply:
The calculation of the size and the call to reserve is done beforehand (line 353) (data_converter -> transform_write)